spec/requests/generic_object_definitions_spec.rb

```diff
@@ -76,6 +76,81 @@
     end
   end
 
+  describe 'GET /api/generic_object_definitions/:id/generic_objects' do
+    it 'can return the generic objects of a definition with appropriate role' do
+      object = FactoryGirl.create(:generic_object, :name => 'bar', :generic_object_definition => object_def)
+      api_basic_authorize subcollection_action_identifier(:generic_object_definitions, :generic_objects, :read, :get)
+
+      get(api_generic_object_definition_generic_objects_url(nil, object_def))
+
+      expected = {
+        'name'      => 'generic_objects',
+        'count'     => 1,
+        'subcount'  => 1,
+        'resources' => [
+          { 'href' => api_generic_object_definition_generic_object_url(nil, object_def, object) }
+        ]
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+
+    it 'does not return the generic objects without an appropriate role' do
+      api_basic_authorize
+
+      get(api_generic_object_definition_generic_objects_url(nil, object_def))
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe 'GET /api/generic_object_definitions/:id/generic_objects/:id' do
+    let(:object) { FactoryGirl.create(:generic_object, :name => 'bar', :generic_object_definition => object_def) }
+    let(:vm) { FactoryGirl.create(:vm_amazon) }
+
+    before do
+      object_def.add_property_attribute('is_something', 'boolean')
```

Comment:
The extensive setup here seems to suggest that the GO API is missing something....perhaps we could at a later point create a factory, or a builder, or at least enhance the factory girl definition to help simplify this process?

Reply:
Yeah, some factories for "simple" objects with dummy attributes would be useful